pkg/agent/util/net.go

```diff
@@ -149,3 +151,136 @@ func GetIPWithFamily(ips []net.IP, addrFamily uint8) (net.IP, error) {
 		return nil, errors.New("no IP found with IPv4 AddressFamily")
 	}
 }
+
+// GetAvailableNodePortIPs gets available NodePort IP addresses with config.
+func GetAvailableNodePortIPs(nodePortIPsFromConfig []string, gateway string) (map[int][]net.IP, map[int][]net.IP, error) {
```

Comment:
IMO this function should be split into 2:
* one function to retrieve all the Node's IPs, which can be defined in this package
* one function to return to validate the list of NodePort IPs provided by the user and return the set of NodePort IPs that can be used, which should not be defined in this package but probably under `cmd/antrea-agent`

Reply:
Done.